test_launch_ros/test/test_launch_ros/actions/test_ros_timer_action.py

```diff
@@ -0,0 +1,201 @@
+
+
+def _shutdown_listener_factory(reasons_arr):
+    return launch.actions.RegisterEventHandler(
+        launch.event_handlers.OnShutdown(
+            on_shutdown=lambda event, context: reasons_arr.append(event)
+        )
+    )
+
+
+def test_timer_action_sanity_check():
+    """Test that timer actions work (sanity check)."""
+    # This test is structured like test_shutdown_preempts_timers
+    # as a sanity check that the shutdown listener
+    # and other launch related infrastructure works as expected
+    shutdown_reasons = []
+
+    ld = launch.LaunchDescription([
+        launch.actions.ExecuteProcess(
+            cmd=[sys.executable, '-c', 'while True: pass'],
+        ),
+
+        RosTimerAction(
+            period=1.,
+            actions=[
+                launch.actions.Shutdown(reason='One second timeout')
+            ]
+        ),
+
+        _shutdown_listener_factory(shutdown_reasons),
+    ])
+
+    ls = launch.LaunchService()
+    ls.include_launch_description(ld)
+    assert 0 == ls.run()
+    assert shutdown_reasons[0].reason == 'One second timeout'
+
+
+def test_shutdown_preempts_timers():
+    shutdown_reasons = []
+
+    ld = launch.LaunchDescription([
+
+        launch.actions.ExecuteProcess(
+            cmd=[sys.executable, '-c', 'while True: pass'],
+        ),
+
+        RosTimerAction(
+            period=1.,
+            actions=[
+                launch.actions.Shutdown(reason='fast shutdown')
+            ]
+        ),
+
+        RosTimerAction(
+            period=2.,
+            actions=[
+                launch.actions.Shutdown(reason='slow shutdown')
+            ]
+        ),
+
+        _shutdown_listener_factory(shutdown_reasons),
+    ])
+
+    ls = launch.LaunchService()
+    ls.include_launch_description(ld)
+    assert 0 == ls.run()
+    assert len(shutdown_reasons) == 1
+    assert shutdown_reasons[0].reason == 'fast shutdown'
+
+
+def test_time_is_passing():
+    """Test that time is actually passing (sanity check)."""
+    # This test starts a 5 second timer and verifies that
+    # at least 5 seconds have passed between start of test
+    # and shutdown
+    shutdown_reasons = []
+
+    ld = launch.LaunchDescription([
+        launch.actions.ExecuteProcess(
+            cmd=[sys.executable, '-c', 'while True: pass'],
+        ),
+
+        RosTimerAction(
+            period=5.,
+            actions=[
+                launch.actions.Shutdown(reason='Five second timeout')
+            ]
+        ),
+
+        _shutdown_listener_factory(shutdown_reasons),
+    ])
+
+    start_time = time.time()
+    ls = launch.LaunchService()
+    ls.include_launch_description(ld)
+    assert 0 == ls.run()
+    assert abs(time.time() - start_time) > 5
+    assert shutdown_reasons[0].reason == 'Five second timeout'
+
+
+def test_timer_uses_sim_time():
+    """Test that timer uses time from /clock topic."""
+    # Create clock publisher node
+    rclpy.init()
+    node = rclpy.create_node('clock_publisher_node')
```

Comment:
nit: I would prefix the test node so it's clear where this came from, in case there's an unclean shutdown and the process lingers
```suggestion
    node = rclpy.create_node('test_ros_timer_action_node')
```